### ISSUE TITLE
Job interview assignment for BackEnd position

### ISSUE THREAD

**Issue by salEZY** [salEZY/Polazak-BE#1]
@polazak

**Comment by polazak** [comment]
Hi @salEZY 

Thanks for your submission. 
First of all let me say sorry for late reply. Github did not send me notification. 

Your code is organized well and easy to read.
I noticed the majority of the code has been committed in the very first commit. So I am not sure how long did it take to complete the full assignment. 

I assume the registration and authentication part of the code was copy-pasted from a previous project, what is not an issue, although it did not matched with swagger API, neither documented in the README.md

I had to tweak your code the be able to test it. I Applied following changes:

```
diff --git a/index.js b/index.js
index c23c36b..f38247d 100644
--- a/index.js
+++ b/index.js
@@ -23,7 +23,8 @@ app.get("/", (req, res) => {
 });
 
 // User routes
-app.use("/api/user", require("./routes/users"));
+// to make it match with swagger
+app.use("/api/auth", require("./routes/users"));
 // Product routes
 app.use("/api/product", require("./routes/product"));
 // Shop-n-Cart routes
diff --git a/routes/users.js b/routes/users.js
index 7bd646a..861f484 100644
--- a/routes/users.js
+++ b/routes/users.js
@@ -16,7 +16,8 @@ router.post(
   [
     check("email").normalizeEmail().isEmail(),
     check("password").isLength({ min: 6 }),
-    check("repeatPassword").isLength({ min: 6 }),
+    // I don't think we need this for an API endpoint:
+    //    check("repeatPassword").isLength({ min: 6 }),
   ],
   async (req, res) => {
     const errors = validationResult(req);
@@ -25,10 +26,10 @@ router.post(
         .status(422)
         .json({ message: "Invalid inputs passed, please check your data" });
 
-    const { email, password, repeatPassword } = req.body;
+    const { email, password /*, repeatPassword */ } = req.body;
 
-    if (password !== repeatPassword)
-      return res.status(422).json({ message: "Passwords do NOT match!" });
+    // if (password !== repeatPassword)
+    //   return res.status(422).json({ message: "Passwords do NOT match!" });
     let user;
     try {
       // Check if user exists
diff --git a/swagger.json b/swagger.json
index 45342d5..f85926c 100644
--- a/swagger.json
+++ b/swagger.json
@@ -84,7 +84,10 @@
             "description": "A list of Products",
             "schema": {}
           }
-        }
+        },
+        "security": [
+          {"APIKeyHeader": []}
+        ]
       }
     },
     "/product/add": {
@@ -261,5 +264,12 @@
         }
       }
     }
+  },
+  "securityDefinitions": {
+    "APIKeyHeader" : {
+      "type": "apiKey",
+      "in": "header",
+      "name": "token"
+    }
   }
 }
```
I tried the `npm run test` but it hangs.

Basically you've done the job. Implemented the business, only missed to double check before submission, and therefore small mistakes here and there made difficult to adopt your code.

Thank you again and I hope you appreciate my honest feedback.

I will inform HR about your coding results, and they will decide.


Cheers,
Oliver

**Comment by salEZY** [comment]
Hi Oliver,

Thank you for the feedback. Yeah i messed up a little...i did write a lot of it prior to first commit, just wanted to have a bare bones working app first. I did start working on it the day i sent the email to HR.

Anyways thanks again.
Cheers,
Aleksandar
